cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/spark/util/CqlEvaluationRequestUtil.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.ibm.cohort.cql.spark.util;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.ibm.cohort.cql.evaluation.CqlEvaluationRequest;
+import com.ibm.cohort.cql.evaluation.parameters.Parameter;
+
+public class CqlEvaluationRequestUtil {
+    private static Map<CqlEvaluationRequest,String> requestToParamsJson = new ConcurrentHashMap<>();
+    
+    public static void clearCache() {
+        requestToParamsJson.clear();
+    }
+    
+    public static String getKeyParametersColumnData(CqlEvaluationRequest request, Set<String> keyParameterNames) {
+        return requestToParamsJson.computeIfAbsent(request, req -> {
+            String parametersJson = "{}";
+            
+            Map<String,Parameter> keyParameters = req.getParameters();
+            if( keyParameters != null ) {
+                if( keyParameterNames != null ) {
+                    keyParameters = keyParameters.entrySet().stream()
+                            .filter( e -> keyParameterNames.contains(e.getKey()) )
+                            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+                }
+                try {
+                    parametersJson = new ObjectMapper().writeValueAsString(keyParameters);
```

Comment:
I think the filtering aspect of this cache setup makes it a bit more confusing
but narrowing the cache `Map` to be keyed on `keyParameters` (instead of the entire `CqlEvaluationRequest`) might be a bit cleaner?

Reply:
I agree there was some code smell here. The cache key did not match up with the parameters declaration. I updated this class so that the key parameters are now an immutable part of the class state and only the request is passed into the getKeyParametersJson method. I also added the functionality related to the rowGroupingDisabled flag. Now, all of the parameters-related logic is in a single place which should address another comment I had received in a separate, offline conversation.